Microsoft.Toolkit.Uwp.UI/Triggers/NetworkConnectionStateTrigger.cs

```diff
@@ -22,13 +25,14 @@ public class NetworkConnectionStateTrigger : StateTriggerBase
         public NetworkConnectionStateTrigger()
         {
             _dispatcherQueue = DispatcherQueue.GetForCurrentThread();
-            var weakEvent =
-                new WeakEventListener<NetworkConnectionStateTrigger, object>(this)
-                {
-                    OnEventAction = (instance, source) => NetworkInformation_NetworkStatusChanged(source),
-                    OnDetachAction = (weakEventListener) => NetworkInformation.NetworkStatusChanged -= weakEventListener.OnEvent
-                };
-            NetworkInformation.NetworkStatusChanged += weakEvent.OnEvent;
+
+            _weakEvent = new WeakEventListener<NetworkConnectionStateTrigger, object, EventArgs>(this)
+            {
+                OnEventAction = (instance, source, eventArgs) => { NetworkInformation_NetworkStatusChanged(source); },
+                OnDetachAction = listener => { NetworkInformation.NetworkStatusChanged -= OnNetworkEvent; }
+            };
+
+            NetworkInformation.NetworkStatusChanged += OnNetworkEvent;
```

Comment:
`NetworkInformation.NetworkStatusChanged += _weakEvent.OnEvent` isn't possible because `NetworkInformation.NetworkStatusChanged` is a custom delgate, not a normal event handler.

Reply:
@puppetsw Let's do the same thing for the `OnDetachAction` if we can.

Reply:
@Arlodotexe I'm sorry, but I'm not sure how to go about this, or I'm not understanding. If you could give a pointer I'm happy to make the change.

Reply:
Dug into this a bit. Looks like `OnDetachAction` doesn't supply the access to the `instance` inside the delegate. Though it does null out the `OnDetachAction` after invoking it, so it should release all references captures by the delegate.

@michael-hawker does that sound right?